src/PhpEnvironment/Request.php

```diff
@@ -490,7 +490,7 @@ protected function detectBaseUrl()
             // matching PHP_SELF.
 
             $argv = $this->getServer()->get('argv', []);
-            if (isset($argv[0]) && strpos($filename, $argv[0]) === 0) {
+            if (!empty($argv[0]) && is_string($argv[0]) && strpos($filename, $argv[0]) === 0) {
```

Comment:
Just simple fix, add space between ! and empty.

Reply:
Maybe you can also UnitTest for it?

Reply:
Is there really a unit test needed? I just made the isset argument more strict. Strpos expects an not empty string as needle and I just ensure, that $argv[0] is always an non empty string.

Reply:
Just to cover the bug/case that it will not happen again after further changes.

Reply:
@webimpress what is your opinion?

Reply:
Hi, see my comment below. I think we need another fix - as argv shouldn't be used for non-CLI requests. And for CLI request, from what I have checked, it's not possible to have empty `argv[0]`.